tests/cases/conformance/expressions/functions/contextuallyTypedIifeStrict.ts

```diff
@@ -14,7 +14,7 @@
 // optional parameters
 ((j?) => j + 1)(12);
 ((k?) => k + 1)();
-((l, o?) => l + o)(12); // o should be any
```

Comment:
This was never true (at least not since this test case was introduced). This was just copy-pasted from the non-strict version of this test (see [here](https://github.com/microsoft/TypeScript/blob/a0e0104656415b6c6d18c6ba1100ae42157a4f09/tests/cases/conformance/expressions/functions/contextuallyTypedIife.ts#L16)) and wasn't removed/adjusted